libc-top-half/musl/src/thread/pthread_create.c

```diff
@@ -558,14 +558,18 @@ int __pthread_create(pthread_t *restrict res, const pthread_attr_t *restrict att
 			__wait(&args->control, 0, 3, 0);
 	}
 #else
+#define WASI_THREADS_MIN_TID 1
+#define WASI_THREADS_MAX_TID 0x1FFFFFFF
 	/* `wasi_thread_spawn` will either return a host-provided thread ID (TID)
-	 * (`>= 0`) or an error code (`< 0`). As in the unmodified version, all
+	 * (`<1, 0x1FFFFFFF>`) or an error code (`< 0`). Please note that `0` is
+	 * reserved for compatibility reasons and must not be returned by the runtime
+	 * (if that happens, EAGAIN is returned). As in the unmodified version, all
```

Comment:
why EAGAIN?
an assertion is more appropriate.

Reply:
Yeah, it might be better to crash to force people to go fix the engine.

Reply:
fair enough, updated.